packages/instantsearch.js/src/types/templates.ts

```diff
@@ -50,7 +50,11 @@ export type TemplateWithBindEvent<TTemplateData = void> =
     ) => VNode | VNode[] | string);
 
 export type Templates = {
-  [key: string]: Template<any> | TemplateWithBindEvent<any> | undefined;
+  [key: string]:
+    | Template<any>
+    | TemplateWithBindEvent<any>
+    | Templates
```

Comment:
don't need this change anymore right?

Reply:
or still for the top-level ones that are not used?

Reply:
We need this to not have a type error with the nested templates